src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ILScanner.cs

```diff
@@ -687,7 +693,11 @@ protected override MethodDesc ResolveVirtualMethod(MethodDesc declMethod, DefTyp
                 return result;
             }
 
-            public override bool CanConstructType(TypeDesc type) => _constructedTypes.Contains(type);
+            public override bool CanReferenceConstructedMethodTable(TypeDesc type)
+                => _constructedMethodTables.Contains(type);
+
+            public override bool CanTypeOrCanonicalFormOfTypeBeAllocated(TypeDesc type)
```

Comment:
These two methods do very similar things, but they have very different names.  Should these two methods have similar names, e.g. `CanReferenceConstructedMethodTable` + `CanReferenceConstructedTypeOrCanonicalFormOfType`; `CanTypeBeAllocated` + `CanTypeOrCanonicalFormOfTypeBeAllocated` ?